src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1529,6 +1529,97 @@ await PreviewBuildIntegratedProjectActionsAsync(buildIntegratedProject, actions,
                 nuGetProjectContext, primarySources, secondarySources, token);
         }
 
+        // Preview and return ResolvedActions for many NuGetProjects.
+        public async Task<IEnumerable<ResolvedAction>> PreviewProjectsInstallPackageAsync(
+            IEnumerable<NuGetProject> nuGetProjects,
+            PackageIdentity packageIdentity,
+            ResolutionContext resolutionContext,
+            INuGetProjectContext nuGetProjectContext,
+            IEnumerable<SourceRepository> activeSources,
+            CancellationToken token)
+        {
+            if (nuGetProjects == null)
+            {
+                throw new ArgumentNullException(nameof(nuGetProjects));
+            }
+
+            if (packageIdentity == null)
+            {
+                throw new ArgumentNullException(nameof(packageIdentity));
+            }
+
+            if (resolutionContext == null)
+            {
+                throw new ArgumentNullException(nameof(resolutionContext));
+            }
+
+            if (nuGetProjectContext == null)
+            {
+                throw new ArgumentNullException(nameof(nuGetProjectContext));
+            }
+
+            if (activeSources == null)
+            {
+                throw new ArgumentNullException(nameof(activeSources));
+            }
+
+            if (!activeSources.Any())
+            {
+                throw new ArgumentException("At least 1 item expected for " + nameof(activeSources));
+            }
+
+            if (packageIdentity.Version == null)
+            {
+                throw new ArgumentNullException(nameof(packageIdentity));
+            }
+
+            var results = new List<ResolvedAction>();
+
+            var buildIntegratedProjectsToUpdate = new List<BuildIntegratedNuGetProject>();
+            var otherTargetProjectsToUpdate = new List<NuGetProject>();
+
+            foreach (var proj in nuGetProjects)
```

Comment:
It would be great to have comments regrading the reason of separating the nugetProjects in two.

Reply:
Added comment about why 2 separate list.